aesara/tensor/elemwise.py

```diff
@@ -1421,17 +1416,21 @@ def perform(self, node, inp, out):
             output[0] = np.array(variable, copy=True, dtype=node.outputs[0].type.dtype)
 
     def infer_shape(self, fgraph, node, shapes):
-        (ishape,) = shapes
-        axis = self.axis
-        if axis is None:
+        (
+            ishape,
+            axis_shape,
+        ) = shapes
+        if axis_shape == 0:
             return ((),)
-        return (
+        axis = node.inputs[1]
+        final_shape = (
             [
                 ishape[i]
                 for (i, b) in enumerate(node.inputs[0].type.broadcastable)
-                if i not in axis
+                if i not in axis.data
```

Comment:
If we're going to do this, we'll need to make sure that `axis` is always a `Constant`.

Reply:
Current approach makes `axis` as a `np.array` instead of `Constant`. Since `Constant` won't be allowed in the `make_node` parameter list.